subsys/net/lib/lwm2m/lwm2m_rw_json.c

```diff
@@ -663,17 +664,19 @@ static int read_int(struct lwm2m_input_context *in, int64_t *value,
 		if (c == '-' && accept_sign && i == 0) {
 			neg = true;
 		} else if (isdigit(c) != 0) {
-			*value = *value * 10 + (c - '0');
+			if (temp > INT64_MAX / 10ULL) {
+				return -EINVAL;
+			}
+
+			temp = temp * 10ULL + (c - '0');
```

Comment:
INT64_MAX = `9223372036854775807` so you're basically allowing `9223372036854775808` and `9223372036854775809` to sneak in (confirmed by testing)

also make sure that when you fix it you still allow `-9223372036854775808` to be a valid input. Currently if the highlighted code were to work as I think you wanted to, it would have not allowed for `9223372036854775808`